pkg/reconciler/kubernetes/tektoninstallerset/install.go

```diff
@@ -225,19 +230,107 @@ func (i *installer) resourceReconcileFields(u *unstructured.Unstructured) []stri
 // this method is written as generic to all the resources
 // currently tested with deployments and StatefulSet
 // TODO: (jkandasa) needs to be tested with other resources too
-func (i *installer) ensureResource(expected *unstructured.Unstructured) error {
+func (i *installer) ensureResource(ctx context.Context, expected *unstructured.Unstructured) error {
 	i.logger.Debugw("verifying a resource",
 		"name", expected.GetName(),
 		"namespace", expected.GetNamespace(),
 		"kind", expected.GetKind(),
 	)
 
-	// update proxy settings
+	// update specific things to deployments and statefulSets
 	if expected.GetKind() == "Deployment" || expected.GetKind() == "StatefulSet" {
+
+		// update proxy settings
 		err := common.ApplyProxySettings(expected)
 		if err != nil {
+			i.logger.Errorw("error on applying proxy settings",
+				"name", expected.GetName(),
+				"namespace", expected.GetNamespace(),
+				"kind", expected.GetKind(),
+				err,
+			)
 			return err
 		}
+
+		// if a deployment or statefulSets managed by HPA, ignore replicas from user input(TektonConfig CR)
+		// and take replicas from HPA status(DesiredReplicas)
+
+		// lists the available HPAs
+		hpaList, err := i.kubeClientSet.AutoscalingV2().HorizontalPodAutoscalers(expected.GetNamespace()).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			i.logger.Error("error on listing HPA", err)
+			return err
+		}
+
+		// check the expected resource configured with HPA
+		var hpa *autoscalingv2.HorizontalPodAutoscaler
+		for _, _hpa := range hpaList.Items {
+			target := _hpa.Spec.ScaleTargetRef
+			if target.Kind == expected.GetKind() && target.Name == expected.GetName() {
+				hpa = _hpa.DeepCopy()
+				break
+			}
+		}
+
+		// if a hpa found to that resource, update replicas count from the HPA "DesiredReplicas"
+		if hpa != nil {
+			desiredReplicas := hpa.Status.DesiredReplicas
+			minReplicas := hpa.Spec.MinReplicas
+
+			i.logger.Debugw("hpa found for this resource, verifying replicas count from hpa",
+				"hpaName", hpa.GetName(),
+				"desiredReplicas", desiredReplicas,
+				"minReplicas", minReplicas,
+				"scaleTargetKind", hpa.Spec.ScaleTargetRef.Kind,
+				"scaleTargetName", hpa.Spec.ScaleTargetRef.Name,
+				"namespace", hpa.GetNamespace(),
+			)
+
+			// if there is no metrics data available in the cluster the HPA desiredReplicas will be zero
+			// compare minReplicas and desiredReplicas and take the higher one
+			if minReplicas != nil && desiredReplicas < *minReplicas {
```

Comment:
a more straightforward way would be to check for `ScalingActive` set to False:
```
  - lastTransitionTime: "2023-09-27T15:57:24Z"
    message: scaling is disabled since the replica count of the target is zero
    reason: ScalingDisabled
    status: "False"
    type: ScalingActive
```

Reply:
updated